src/app/clusters/basic-information/basic-information.cpp

```diff
@@ -49,6 +51,21 @@ using chip::Protocols::InteractionModel::Status;
 
 namespace {
 
+class NodeConfigurationListenerImpl : public NodeConfigurationListener
+{
+public:
+    void OnConfigurationVersionChanged() override
+    {
+        MatterReportingAttributeChangeCallback(0, BasicInformation::Id, BasicInformation::Attributes::ConfigurationVersion::Id);
+    }
+
+    static NodeConfigurationListenerImpl & Instance()
```

Comment:
Using a singleton here is not the right approach. Init of the cluster should do the registration.

Reply:
I have tried to rework this, PTAL.